lib/widgets/dialog.dart

```diff
@@ -1,20 +1,32 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 
 import '../generated/l10n/zulip_localizations.dart';
 import 'actions.dart';
 
-Widget _dialogActionText(String text) {
-  return Text(
-    text,
-
-    // As suggested by
-    //   https://api.flutter.dev/flutter/material/AlertDialog/actions.html :
-    // > It is recommended to set the Text.textAlign to TextAlign.end
-    // > for the Text within the TextButton, so that buttons whose
-    // > labels wrap to an extra line align with the overall
-    // > OverflowBar's alignment within the dialog.
-    textAlign: TextAlign.end,
-  );
+/// A platform-appropriate action for [AlertDialog.adaptive]'s [actions] param.
+Widget _adaptiveAction({required VoidCallback onPressed, required String text}) {
+  switch (defaultTargetPlatform) {
+    case TargetPlatform.android:
+    case TargetPlatform.fuchsia:
+    case TargetPlatform.linux:
+    case TargetPlatform.windows:
+      return TextButton(
+        onPressed: onPressed,
+        child: Text(
+          text,
+          // As suggested by
+          //   https://api.flutter.dev/flutter/material/AlertDialog/actions.html :
+          // > It is recommended to set the Text.textAlign to TextAlign.end
+          // > for the Text within the TextButton, so that buttons whose
+          // > labels wrap to an extra line align with the overall
+          // > OverflowBar's alignment within the dialog.
+          textAlign: TextAlign.end));
+    case TargetPlatform.iOS:
```

Comment:
nit: a blank line to separate these cases helps make the structure easier to see:
```suggestion
          textAlign: TextAlign.end));

    case TargetPlatform.iOS:
```